<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/cami2-assembly/tutorial.html" />
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/" />

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material" />
<meta name="DC.type" content="text" />
<meta name="DC.title" content="Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy" />
<meta name="DC.publisher" content="Galaxy Training Network" />
<meta name="DC.date" content="2022-03-27 09:19:20 +0000" />
<meta name="DC.creator" content="PlushZ" />
<meta name="DC.creator" content="Bérénice Batut" />

        
        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:site_name" content="Galaxy Training Network" />
        <meta property="og:title" content="Galaxy Training: Reproducing Critical Assessment of Metagenome Interpretat..." />
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />

    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/cami2-assembly/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/cami2-assembly/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/cami2-assembly/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/cami2-assembly/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/cami2-assembly/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/cami2-assembly/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/cami2-assembly/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/cami2-assembly/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-03-01/topics/metagenomics/tutorials/cami2-assembly/tutorial.html" title="Version 2022-03-01">2022-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-02-01/topics/metagenomics/tutorials/cami2-assembly/tutorial.html" title="Version 2022-02-01">2022-02-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-01-01/topics/metagenomics/tutorials/cami2-assembly/tutorial.html" title="Version 2022-01-01">2022-01-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2022-03-27 09:19:20 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/cami2-assembly/tutorial.html",
  "timeRequired": "PT40H00M",
  "keywords": "metagenomics, assembly",
  "description": "The questions this  addresses are:\n - How to reproduce CAMI challenge?\n - How to add/update tool in Galaxy?\n - How to benchmark results of CAMI challenge?\n\n\nThe objectives are:\n - Be familiar with CAMI challenge\n - Be able to select one challenge and benchmarking datasets\n - Be able to upload benchmarking datasets into Galaxy\n - Be familiar with Galaxy computational resources\n - Be familiar with existing tools in Galaxy\n - Be familiar with databases / reference genomes that are available in Galaxy\n - Be able to add/update the tool in Galaxy\n - Be able to choose assembling tool based on dataset metadata\n - Perform assembly with Flye, Megahit, Abyss, MetaSPAdes tools\n - Produce a benchmarking analysis of these assemblies with Quast, Bowtie2, Map with Minimap2, Samtools, and MultiQC tools\n - Create plots with Python to compare results\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/PlushZ/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/PlushZ/",
      "name": "PlushZ",
      "image": "https://avatars.githubusercontent.com/PlushZ",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/PlushZ/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/PlushZ/",
      "name": "PlushZ",
      "image": "https://avatars.githubusercontent.com/PlushZ",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ]
}
</script>

<section class="tutorial topic-metagenomics">
    <h1 data-toc-skip>Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy</h1>
    

    <div markdown="0">

	<div class="contributors-line">
		Authors: 
<a href="/training-material/hall-of-fame/PlushZ/" class="contributor-badge contributor-PlushZ"><img src="https://avatars.githubusercontent.com/PlushZ?s=27" alt="Avatar">PlushZ</a>



<a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/bebatut?s=27" alt="Avatar">Bérénice Batut</a>



	</div>

</div>


    <blockquote class="overview">
        <div class="box-title" aria-label="overview box">Overview</div>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to reproduce CAMI challenge?</p>
</li>
        
        <li><p>How to add/update tool in Galaxy?</p>
</li>
        
        <li><p>How to benchmark results of CAMI challenge?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Be familiar with CAMI challenge</p>
</li>
        
        <li><p>Be able to select one challenge and benchmarking datasets</p>
</li>
        
        <li><p>Be able to upload benchmarking datasets into Galaxy</p>
</li>
        
        <li><p>Be familiar with Galaxy computational resources</p>
</li>
        
        <li><p>Be familiar with existing tools in Galaxy</p>
</li>
        
        <li><p>Be familiar with databases / reference genomes that are available in Galaxy</p>
</li>
        
        <li><p>Be able to add/update the tool in Galaxy</p>
</li>
        
        <li><p>Be able to choose assembling tool based on dataset metadata</p>
</li>
        
        <li><p>Perform assembly with Flye, Megahit, Abyss, MetaSPAdes tools</p>
</li>
        
        <li><p>Produce a benchmarking analysis of these assemblies with Quast, Bowtie2, Map with Minimap2, Samtools, and MultiQC tools</p>
</li>
        
        <li><p>Create plots with Python to compare results</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 40 hours 00 minutes</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/metagenomics/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            

            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/metagenomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 27, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#select-the-challenge-to-reproduce" id="markdown-toc-select-the-challenge-to-reproduce">Select the challenge to reproduce</a></li>
  <li><a href="#select-and-get-the-data-to-benchmark" id="markdown-toc-select-and-get-the-data-to-benchmark">Select and get the data to benchmark</a></li>
  <li><a href="#select-and-prepare-the-tools-to-run" id="markdown-toc-select-and-prepare-the-tools-to-run">Select and prepare the tools to run</a></li>
  <li><a href="#assess-the-assembly-outputs" id="markdown-toc-assess-the-assembly-outputs">Assess the assembly outputs</a></li>
</ol>

</blockquote>

<h2 class="no_toc" id="introduction">Introduction</h2>

<h3 id="global-context">Global context</h3>

<p>Metagenomics analysis involves complex computational methods like assembly, genome binning, taxonomic binning, taxonomic profiling and other. It is only when these initial data processing methods make sense that <span class="notranslate">downstream</span> analyses and information extraction are meaningful. Even though vast progress has been made over the last few years, none of these approaches can fully recover the complex information encoded in metagenomes. These approaches all rely on simplifying assumptions that can have serious limitations.</p>

<p>It is usually difficult for microbiome data analysts to know which tools to use for each step. They must rely on evaluations made by the developers of new or improved methods. However, these evaluations are rarely comparable: there is no general standard for the evaluation of computational methods in metagenomics. As a result, users may not be properly informed and compute predictions may be misinterpreted.</p>

<h3 id="the-cami-challenge">The CAMI Challenge</h3>

<p>Essentially, the critical assessment concept involves evaluating a theory, situation, statement, or something else with the goal of supporting its dominant paradigms or disproving them as well as suggesting a better alte<span class="notranslate">rna</span>tive plan. Any view or conclusion needs to be backed up by credible evidence in order to be considered critical. In metagenomic research, critical assessment of the data interpretation is especially important, because merely accepting the data as truth would not suffice.</p>

<p><span class="citation"><a href="#noauthor_critical_nodate">“<b>Critical Assessment of Metagenome Interpretation—a benchmark of metagenomics software \textbar Nature Methods</b>”</a></span> (CAMI) was established in 2012. The objective of CAMI is to evaluate metagenomics methods independently, comprehensively, and without bias. The initiative provides users with comprehensive information about the performance of methods in all relevant scenarios. Therefore, it assists users in the selection and application of methods as well as their proper interpretation.</p>

<p>During the 1st CAMI challenge, extensive metagenome benchmark data sets were generated from newly sequenced genomes of around 700 microbial isolates and approximately 600 circular elements that were distinct from strains, species, genera or orders represented by public genomes. Four challenges were suggested (assembly, binning, taxonomic profiling, taxonomic binning). Overall, 16 teams worldwide submitted 215 submissions to the CAMI1 challenge, consisting of 25 programs and 36 biobox implementations, with permission to publish.</p>

<p><span class="citation"><a href="#noauthor_critical_nodate-1">“<b>Critical Assessment of Metagenome Interpretation - the second round of challenges \textbar bioRxiv</b>”</a></span> (CAMI2) started in 2019. It includes 4 different challenges:</p>

<ul>
  <li>An assembly challenge</li>
  <li>A genome binning challenge</li>
  <li>A taxonomic binning challenge</li>
  <li>A taxonomic profiling challenge</li>
  <li>Clinical pathogen prediction challenge</li>
</ul>

<p>Each challenge uses a similar set of benchmark datasets reproducing different environments and different sequencing technologies  long and short <span class="notranslate">reads</span>). The six benchmark datasets—reflecting a range of complexities—were created from 1,680 microbial genomes and 599 circular elements of viruses and plasmid:</p>

<ul>
  <li>2 ‘toy’ datasets created from public data and provided before a challenge,</li>
  <li>3 ‘challenge’ datasets derived exclusively from genomic data that were not publicly available at the time
    <ul>
      <li>Marine
        <ul>
          <li>Simulated short read and long read shotgun metagenome data from samples at different seafloor locations of a marine environment</li>
        </ul>
      </li>
      <li>Plant-associated
        <ul>
          <li>Simulated short read and long read shotgun metagenome data from samples taken from a plant rhizosphere environment</li>
        </ul>
      </li>
      <li>strain madness
        <ul>
          <li>Simulated long read and short read shotgun metagenome data, including a large amount of strain-level variation</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>a pathogen detection challenge dataset was offered, based on a clinical metagenome sample of blood from a critically ill patient with hemorrhagic fever of unknown cause.</li>
</ul>

<p>During the challenges, Datasets for the challenge were available for download only to participants via the <a href="https://data.cami-challenge.org/">CAMI portal</a>. After the challenges, all CAMI benchmark datasets were made available with digital object identifiers (DOIs) (<a href="https://www.nature.com/articles/s41596-020-00480-3/tables/2">Table 2</a>) and the genomic data are now in public sequence repositories such as the National Center for Biotechnology Information (NCBI) to be used for further benchmarking in the field.</p>

<p>Different tools were run with different parameters on the various datasets. Participants ran their preferred tools with their preferred set of parameters and  submitted their results along with either a Docker container containing the complete <span class="notranslate">workflow</span>, a bioconda script or a software repository with detailed installation instructions, specifying all parameter settings and reference databases used. A total of 5,002 submissions were received from 30 exte<span class="notranslate">rna</span>l teams and CAMI developers for the four challenge datasets. Following that, the CAMI developers evaluated the results using standardised metrics and then made sense of the different results described in <a href="https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1">Meyer et al, 2020</a></p>

<p><span class="notranslate">Galaxy</span> could be considered as a good platform for such a challenge as CAMI.</p>

<p>In addition to providing the computational resources, <span class="notranslate">Galaxy</span> provides full reproducibility, which makes it a great solution for CAMI and similar types of challenges. Everything required to complete such challenges is collected in one location:</p>

<ul>
  <li>Datasets which can be uploaded to data libraries and shared with public</li>
  <li>Databases and tools of different versions can be available to use</li>
  <li>History of tool runs with documented parameters of tool runs, inputs, outputs, command line, job parameters (CPU, memory usage, runtime, etc.)</li>
  <li><span class="notranslate">Galaxy</span> workflows can be extracted and shared for the reproducibility</li>
  <li>Etc.</li>
</ul>

<p>In <span class="notranslate">Galaxy</span>, users can choose to share their history with certain users or to publish it so that everyone can access the history.</p>

<h3 id="motivation">Motivation</h3>

<p>In this tutorial we show how <span class="notranslate">Galaxy</span> could be used as a platform to support the next CAMI challenges or for other similar critical assessment challenges, but also what were the issues we got and possible solutions (on <span class="notranslate">Galaxy</span> side but for organisers of such challenges).</p>

<h2 id="select-the-challenge-to-reproduce">Select the challenge to reproduce</h2>

<p>In this section we  choose the challenge to reproduce. As the entire challenge is not a one-day task and takes months to complete, we should focus on one challenge out of four - assembly, genome binning, taxonomic binning, or taxonomic profiling. The choice we make depends on our interests and objectives.</p>

<h3 id="description-of-cami-challenges">Description of CAMI challenges</h3>

<h4 id="assembly-challenge">Assembly challenge</h4>

<p>Assembly is one of the important components of metagenomic analysis. The challenge is called to assess the quality and performance of tools for assembling strain-resolved genomes.</p>

<p>Three “challenge” datasets were offered:</p>

<ul>
  <li>marine,</li>
  <li>strain madness, and</li>
  <li>plant associated data.</li>
</ul>

<p>Two types of assemblies were assessed: single-sample assembly, i.e. assembly of <span class="notranslate">reads</span> from only one sample at a time, and co-assembly, i.e. assembly of all <span class="notranslate">reads</span> from several samples together. For the sake of comparison, gold standard assemblies, i.e. the correct result which can be used to compare with and benchmark other methods’ results, were created with all regions covered by at least one read and were provided only after the challenges.</p>

<p>Participants provided 155 submissions and 20 assembler versions of 10 assembling tools to be evaluated during this challenge. The different tools performed differently while measuring various metrics. By CAMI2 developers there are provided <a href="https://www.biorxiv.org/content/biorxiv/early/2021/07/12/2021.07.12.451567/F1.large.jpg?width=800&amp;height=600&amp;carousel=1">plots</a> of benchmarking analyses of different tools depending on different statistics.</p>

<p><img src="../../images/cami2/assembly-tools-plots.jpg" alt="Assembly tools plot from CAMI2. " width="70%" loading="lazy" /></p>

<p>Overall, GATB<a href="https://pubmed.ncbi.nlm.nih.gov/24990603/">[3]</a> ranked best on the strain madness data across most metrics (mismatches, misassemblies, strain recall, and strain precision), while HipMer<a href="https://ieeexplore.ieee.org/abstract/document/8665813">[4]</a> <a href="https://ieeexplore.ieee.org/abstract/document/7832788">[5]</a> <a href="https://www.nature.com/articles/s41598-020-67416-5">[6]</a> on the plant-associated data. Compared to the first CAMI challenge A-STAR<a href="https://www.a-star.edu.sg/"> [7]</a> considerably improved in genome fraction for the strain madness data. The other leader SPAdes <a href="https://pubmed.ncbi.nlm.nih.gov/22506599/">[8]</a> <a href="https://genome.cshlp.org/content/27/5/824?ijkey=fef5a6b24fb0024b39509480f85afaa81724d3e8&amp;keytype2=tf_ipsecsha">[9]</a> was not introduced in the CAMI1 challenge <a href="https://www.nature.com/articles/nmeth.4458">[1]</a> and performed well in CAMI2. For the type of assembly, the different tools performed differently. Single-sample assembly was done well by all assemblers. HipMer <a href="https://ieeexplore.ieee.org/abstract/document/8665813">[4]</a> <a href="https://ieeexplore.ieee.org/abstract/document/7832788">[5]</a> <a href="https://www.nature.com/articles/s41598-020-67416-5">[6]</a> performed better on single samples as well as for pooled samples.</p>

<h4 id="genome-binning-challenge">Genome binning challenge</h4>

<p>Genome binning  clusters or classifies sequences or contigs into bins representing genomes.</p>

<p>It takes as input the output of assembly. Two types of short read assemblies were given as inputs for CAMI2 <a href="https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1">[2]</a>: MEGAHIT <a href="https://www.biorxiv.org/lookup/external-ref?access_num=25609793&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F07%2F12%2F2021.07.12.451567.atom">[10]</a> assembly and Gold Standard Assembly. Both types for strain madness, marine, and plant-associated datasets, so 6 possible datasets in total.</p>

<p>Participants submitted 95 results and 10 binning tools (overall, 18 binner versions). The results were evaluated based on .</p>

<ul>
  <li>the average purity of bins,
    <ul>
      <li>The average purity is the fraction of correctly assigned base pairs for all assignments to a given bin averaged over all predicted genome bins, where unmapped genomes are not considered.</li>
      <li>This value reflects how trustworthy the bin assignments are on average.</li>
    </ul>
  </li>
  <li>completeness of genomes recovered,</li>
  <li>Adjusted Rand Index which quantifies binning performance for the overall dataset.
    <ul>
      <li>According to the <a href="https://www.biorxiv.org/lookup/google-scholar?link_type=googlescholar&amp;gs_type=article&amp;author[0]=F.+Meyer&amp;title=AMBER:+Assessment+of+Metagenome+BinnERs&amp;publication_year=2018&amp;journal=Gigascience&amp;volume=7">paper</a> <a href="https://academic.oup.com/gigascience/article/7/6/giy069/5034950?login=true">[11]</a>, genome binners generate groups or clusters of <span class="notranslate">reads</span> and contigs for a given dataset. Instead of calculating performance metrics established with a bin-to-genome <span class="notranslate">mapping</span>, the quality of a clustering can be evaluated by measuring the similarity between the obtained and correct cluster partitions of the dataset, corresponding here to the predicted genome bins and the gold standard contig or read genome assignments, respectively. Two contigs or <span class="notranslate">reads</span> of the same genome that are placed in the same predicted genome bin are considered true positives TP. Two contigs or <span class="notranslate">reads</span> of different genomes that are placed in different bins are considered true negatives TN. The Rand index ranges from 0 to 1 and is the number of true pairs, TP + TN, divided by the total number of pairs. However, for a random clustering of the dataset, the Rand index would be larger than 0. The adjusted Rand index (ARI) corrects for this by subtracting the expected value for the Rand index and normalising the resulting value, such that the values still range from 0 to 1.</li>
    </ul>
  </li>
</ul>

<p>The performance of different genome binning tools differed depending on metrics, datasets, and assembly type. Overall, the best trade-off performances were given by MetaBinner<a href="https://www.biorxiv.org/content/10.1101/2021.07.25.453671v1">[12]</a> on marine and strain madness assemblies, and CONCOCT <a href="https://www.biorxiv.org/lookup/external-ref?access_num=10.1038/nmeth.3103&amp;link_type=DOI">[13]</a> on plant-associated assemblies.</p>

<h4 id="taxonomic-binning-challenge">Taxonomic binning challenge</h4>

<p>Taxonomic binners group sequences into bins labelled with a taxonomic identifier.. For the challenge,  input datasets were marine, strain madness, and plant-associated of 2 types: <span class="notranslate">reads</span> or gold standard assemblies.</p>

<p>547 results for 7 tools were submitted and evaluated using   3 metrics :</p>

<ul>
  <li>average purity,</li>
  <li>completeness of bins, and</li>
  <li>the accuracy per sample (the fraction of contigs, or base pairs, that have been assigned by a method to the correct taxa for a taxonomic rank).</li>
</ul>

<p>Performances for all datasets decreased from genus to species for all metrics, most notably for completeness by 22.2%.</p>

<p>Across all datasets and all metrics the best performance was shown by MEGAN <a href="https://www.biorxiv.org/lookup/external-ref?access_num=10.1371/journal.pcbi.1004957&amp;link_type=DOI">[14]</a>, closely followed by Kraken <a href="https://www.biorxiv.org/lookup/google-scholar?link_type=googlescholar&amp;gs_type=article&amp;author[0]=D.%20E.+Wood&amp;author[1]=J.+Lu&amp;author[2]=B+Langmead&amp;title=Improved+metagenomic+analysis+with+Kraken+2&amp;publication_year=2019&amp;journal=Genome+Biol&amp;volume=20">[15]</a> v.2.0.8-beta. However, in terms of certain metrics different binners performed better.</p>

<p>There was a post-processing data approach used to improve the results. This approach means filtering of the 1% smallest predicted bins per taxonomic level. Filtering increased purity whereas reduced completeness. Accuracy was not affected. The winner MEGAN was not changed though.</p>

<h4 id="taxonomic-profiling-challenge">Taxonomic profiling challenge</h4>

<p>Using metagenome samples, taxonomic profilers identify and quantify microbial community taxa. There were predicted taxonomic identities and relative abundances of microbial community members for the 64 samples of the mouse gut dataset. There were 4195 results for 13 tools evaluated. The majority of results were from short-read samples and a few from long-read.</p>

<p>The performance of the tools were evaluated using:</p>

<ul>
  <li>purity of identified taxa,</li>
  <li>taxon abundance,
    <ul>
      <li>For quantifying relative abundance estimates, the L1 norm and weighted UniFrac error are determined. The L1 norm assesses relative abundance estimates of taxa at a taxonomic rank, on the basis of the sum of the absolute differences between the true and predicted abundances across all taxa. The weighted UniFrac error uses a taxonomic tree storing the predicted abundances at the appropriate nodes for eight major taxonomic ranks. The UniFrac error is the total amount of predicted abundances that must be moved along the edges of the tree to cause them to overlap with the true relative abundances.</li>
    </ul>
  </li>
  <li>alpha diversity of profiling results.</li>
</ul>

<p>There was a notable performance drop observed from genus to species level rank for all datasets: marina, strain madness, and plant-associated.</p>

<p>mOTUs <a href="https://www.biorxiv.org/lookup/external-ref?access_num=10.1038/s41467-019-08844-4&amp;link_type=DOI">[16] </a>v.2.5.1 performed the best on the marine data at both genus and species. For strain madness data, leaders for genus rank were MetaPhlAn <a href="https://www.nature.com/articles/nmeth.2066">[17]</a> <a href="https://elifesciences.org/articles/65088">[18]</a> v.2.9.2, MetaPhyler <a href="https://www.biorxiv.org/lookup/google-scholar?link_type=googlescholar&amp;gs_type=article&amp;author[0]=B.+Liu&amp;author[1]=T.+Gibbons&amp;author[2]=M.+Ghodsi&amp;author[3]=T.+Treangen&amp;author[4]=M+Pop&amp;title=Accurate+and+fast+estimation+of+taxonomic+profiles+from+metagenomic+shotgun+sequences&amp;publication_year=2011&amp;journal=BMC+Genomics&amp;volume=12">[19]</a> v.1.2571, and mOTUs of the CAMI1. For plant-associated data Bracken v2.6 performed best for completeness across ranks.</p>

<p>Abundances across ranks and submissions were on average predicted better for strain madness than marine data. On the marine data, mOTUs v.2.5.1 had the lowest L1 error while mOTUs v.2.5.1 and MetaPhlAn v.2.9.22 both had the lowest UniFrac error.</p>

<p>On the strain madness data, mOTUs cami1 performed best in L1 norm error whilst MetaPhlAn v.2.9.22 had the lowest UniFrac error.</p>

<p>On the plant-associated data, Bracken v.2.6 had the lowest L1 norm error across ranks and had the lowest UniFrac as “sourmash gather 3.3.2 k31” did.</p>

<h3 id="compare-tools-used-in-cami2-with-what-is-already-exist-in-galaxy">Compare tools used in CAMI2 with what is already exist in <span class="notranslate">Galaxy</span></h3>

<p>We compared tools submitted for every challenge to understand the start point of CAMI2 reproduction in the <span class="notranslate">Galaxy</span>. We collected metadata for each tool:</p>

<ul>
  <li>source code,</li>
  <li>Publication that describes the tool</li>
  <li>Tool version</li>
  <li>Availability in <span class="notranslate">Galaxy</span></li>
  <li>owner/ developer in <span class="notranslate">Galaxy</span></li>
  <li><span class="notranslate">Galaxy</span> wrapper directory</li>
  <li>Tool version in <span class="notranslate">Galaxy</span></li>
  <li>Relevance in <span class="notranslate">Galaxy</span></li>
  <li>Availability in conda-forge / bioconda</li>
  <li>Notes on their performance</li>
  <li>Ranking according to CAMI2 results</li>
</ul>

<blockquote class="details">
  <h3 id="details-cami2-assembly-tools-comparison"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> CAMI2 Assembly tools comparison</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=0&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<blockquote class="details">
  <h3 id="details-cami2-genome-binning-tools-comparison"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> CAMI2 Genome binning tools comparison</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=836882193&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<blockquote class="details">
  <h3 id="details-cami2-taconomic-binning-tools-comparison"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> CAMI2 Taconomic binning tools comparison</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=794114188&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<blockquote class="details">
  <h3 id="details-cami2-taxonomic-profiling-tools-comparison"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> CAMI2 Taxonomic profiling tools comparison</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=2127016351&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>After this comparison it turned out that the Assembly challenge is more interesting for us to reproduce on <span class="notranslate">Galaxy</span> platform. One reason for this choice is that highly ranked tools in CAMI2 (like Megahit, Flye, MetaSPAdes, etc.) are already available in <span class="notranslate">Galaxy</span>, so that we can focus more on benchmarking analysis of assembly results than on adding missing tools into <span class="notranslate">Galaxy</span>.</p>

<h2 id="select-and-get-the-data-to-benchmark">Select and get the data to benchmark</h2>

<p>Once we have decided to reproduce the assembly challenge, we can move forward to determine which dataset is more representative for our purposes. There is no need to use all 3 benchmark datasets (marine, strain madness, plant-associated) or 2 toy datasets in order to demonstrate the utility of <span class="notranslate">Galaxy</span> for future CAMI and other similar tasks.</p>

<h3 id="datasets-description">Datasets description</h3>

<h5 id="toy-datasets">Toy datasets</h5>

<p>As in the first CAMI challenge metagenome “toy” (or “practice”) benchmark datasets were created from public genomes and provided together with the standard of truth before the challenges, to enable contest participants to familiarise themselves with data types and formats. These included</p>

<ul>
  <li>a 49-sample simulated metagenome data from Human Microbiome data to represent five different body sites of the human host, namely gastrointestinal tract, oral cavity, airways, skin and urogenital tract.</li>
</ul>

<blockquote class="details">
  <h3 id="details-human-microbiom-dataset-details"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Human microbiom dataset details</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=503333420&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<ul>
  <li>a 64-sample simulated metagenome data from the guts of different mice, vendors and positions in the gut. (<strong><a href="https://openstack.cebitec.uni-bielefeld.de:8080/swift/v1/CAMISIM_MOUSEGUT">https://openstack.cebitec.uni-bielefeld.de:8080/swift/v1/CAMISIM_MOUSEGUT</a></strong>)
    <ul>
      <li>According to the tutorial <a href="https://www.nature.com/articles/s41596-020-00480-3">[20]</a> the mouse gut metagenome toy dataset was generated with CAMISIM v.0.2 <a href="https://pubmed.ncbi.nlm.nih.gov/30736849/">[21]</a> using a microbial community genome abundance distribution modelled from 791 public prokaryotic genomes marked as at least ‘scaffolds’ in the NCBI RefSeq <a href="https://pubmed.ncbi.nlm.nih.gov/17130148/">[22]</a>. They comprise 8 phyla, 18 classes, 26 orders, 50 families, 157 genera, and 549 species. The community genome abundance distribution matches as closely as possible the 16S taxonomic profiles for 64 mouse gut samples. As such, this dataset enables us to assess how well sequenced community members can be characterised with different techniques from the metagenomes of similar communities. On average, within each of the 64 samples, 91.8 genomes are represented. Both long- and short-read metagenomic sequencing data are available. The runtime to generate this data was approximately 3 weeks using eight CPU cores of a computer with an AMD Opteron 6378 CPU and 968 GB of main memory.</li>
    </ul>
  </li>
</ul>

<p>Both include  5 Gb long paired-end <span class="notranslate">reads</span> (Pacific Biosciences, variable length with a mean of 3000 bp) and 5 Gb short paired-end <span class="notranslate">reads</span> (Illumina HiSeq2000, 150 bp).</p>

<h5 id="challenge-datasets">Challenge datasets</h5>

<p>For the assembly challenge itself, 4 “challenge” datasets were offered:</p>

<ul>
  <li>clinical pathogen detection,</li>
  <li>marine,</li>
  <li>strain madness,</li>
  <li>plant associated data.</li>
</ul>

<h6 id="clinical-pathogen-detection">Clinical Pathogen detection</h6>

<p>The clinical pathogen detection dataset, a short-read metagenomic sequencing dataset of a blood sample from a patient with hemorrhagic fever of unknown cause, was excluded because it is provided only for the last challenge: identify a causal pathogen together with further pathogens.</p>

<blockquote class="details">
  <h3 id="details-clinical-pathogen-detection-dataset-links"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Clinical Pathogen detection dataset links</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=1317732159&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<h6 id="marine">Marine</h6>

<p>The dataset represents a marine environment.</p>

<blockquote class="details">
  <h3 id="details-marine-dataset-details"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Marine dataset details</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=1420005524&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>The dataset represents a marine environment. It was created with CAMISIM from BIOM profiles of a deep-sea environment, using 155 newly sequenced marine isolate genomes from this environment and 622 public genomes with matching taxonomic provenance from MarRef, a manually curated database with completely sequenced marine genomes. Additionally, 200 newly sequenced circular elements including plasmids and viruses were added.</p>

<p>The dataset (100Gb) consists of 2x10 samples (5 Gb each) paired-end short (Illumina, with the length of 150bp) and long-read (Nanopore, with the length of 7408 bp) sequences.</p>

<h6 id="strain-madness">Strain madness</h6>

<blockquote class="details">
  <h3 id="details-strain-madness-dataset-links"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Strain madness dataset links</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=548769407&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>The strain madness dataset represents a very high strain diversity environment (“strain madness”) generated with CAMISIM, using 408 newly sequenced genomes. Dataset (400 Gb) includes 2x100 samples (2 Gb each) paired-rend short (Illumina, the length of 150bp) and long(Nanopore with the length of 7408 bp) read sequences.</p>

<h6 id="plant-associated">Plant-associated</h6>

<blockquote class="details">
  <h3 id="details-plant-associated-dataset-links"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Plant-associated dataset links</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=934346728&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>The dataset represents a plant-associated environment (that included fungal genomes and host plant material). It includes 894 genomes:</p>

<ul>
  <li>224 are from the proGenomes terrestrial representative genomes,</li>
  <li>216 are newly sequenced genomes from an <em>Arabidopsis thaliana</em> root rhizosphere,</li>
  <li>55 are fungal genomes associated with the rhizosphere,</li>
  <li>398 are plasmids or circular elements and one <em>Arabidopsis thaliana</em> genome.</li>
</ul>

<p>15.3% (137) of these genomes have at least one closely related genome present. 90% of metagenome sequence data originate from bacterial genomes, 9% are fungal genome sequences, and 1% is from <em>A. thaliana</em>. 399 circular elements including plasmids and viruses were added.</p>

<p>To evaluate the assembly quality of single-sample versus cross-assembly strategies, 23 new genomes from eight clusters of closely related genomes were selected and added to the dataset in certain samples with predetermined abundances.</p>

<p>Dataset (315Gb) includes 3x21 (5 Gb each paired-end read) paired-end short (with the length of 150bp) and long (with the length of 7408 bp or 3000 bp (sd 1000 bp)) read sequences</p>

<h4 id="dataset-choice">Dataset choice</h4>

<p>We analysed all datasets and compared them with each other in terms of number of samples, dataset size, average read length.</p>

<blockquote class="details">
  <h3 id="details-datasets-comparison"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Datasets comparison</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=650574223&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<ol>
  <li>After close consideration of all datasets we decide to choose a marine dataset:It contains paired-end <span class="notranslate">reads</span> from 10 samples, both short (sequenced with Illumina method) and long (sequenced with nanopore method). ITmeans that we can try different tools on either one sample or pooled sample collection and also rub tools working with long <span class="notranslate">reads</span> and short <span class="notranslate">reads</span>, and tools working with both short and long at the same time.</li>
  <li>Overall size of all marine data is 180 Gb. This size is big enough and allows us to test tools and measure such metrics as runtime and memory usage. Other datasets (strain madness and plant-associated) are too large, and since we have a limit in time of this tutorial we decided to give preference to smaller but highly representable marine dataset.</li>
</ol>

<h3 id="get-datasets">Get datasets</h3>

<p>So the assembly challenge is our choice of challenge that we are interested in. And marine dataset is our choice of dataset that we want to launch an assembly challenge on. Let’s start with the first step - uploading data to <span class="notranslate">Galaxy</span>.</p>

<h4 id="download-data-from-cami2-resources">Download data from CAMI2 resources.</h4>

<p>Since data is of large size we cannot just download them to our computer by clicking the button download. Additionally, data cannot be downloaded at once, datasets are divided by samples - one file per one sample. There are also different files for long and short <span class="notranslate">reads</span> of the same sample.</p>

<p>It is possible to download datasets using DOI links from <a href="https://www.nature.com/articles/s41596-020-00480-3/tables/2">Table 2</a> CAMI Tutorial <a href="https://www.nature.com/articles/s41596-020-00480-3">[20]</a></p>

<table>
  <thead>
    <tr>
      <th>CAMI benchmark dataset</th>
      <th>DOI</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>CAMI I: low, medium, high complexity, and ‘toy’ datasets</td>
      <td>10.5524/100344</td>
    </tr>
    <tr>
      <td>CAMI II: human microbiome project and mouse gut toy datasets</td>
      <td>10.4126/FRL01-006425518; 10.4126/FRL01-006421672</td>
    </tr>
    <tr>
      <td>CAMI II: marine, strain madness, rhizosphere, and pathogen detection</td>
      <td>10.4126/FRL01-006425521</td>
    </tr>
  </tbody>
</table>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-dataset-with-doi"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get dataset with DOI</h3>

  <ol>
    <li>Use the table above to find unique identifier (DOI) of dataset you need</li>
    <li>Use <a href="https://search.datacite.org/">https://search.datacite.org/</a> to search for data via DOI, and click the link that was found for the DOI.</li>
    <li>On the page of dataset there is the link provided to download dataset -  for marine dataset all samples available via <a href="https://frl.publisso.de/data/frl:6425521/marine/">https://frl.publisso.de/data/frl:6425521/marine/</a></li>
    <li>Download datasets on your computer</li>
  </ol>
</blockquote>

<blockquote class="details">
  <h3 id="details-download-datasets-with-cami-client"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Download datasets with CAMI Client</h3>

  <p>Another way to download datasets is to use CAMI Client. <a href="https://www.microbiome-cosi.org/cami/resources/cami-client">Here</a> you can find instructions on how to use it.</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on-get-dataset-with-doi-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get dataset with DOI</h3>

    <ol>
      <li>Install Java 8 on your computer.</li>
      <li>Create an account on<a href="https://data.cami-challenge.org/"> CAMI portal</a></li>
      <li>Download the <a href="https://data.cami-challenge.org/camiClient.jar">camiClient.jar</a></li>
      <li>Get text-file with links for dataset from <a href="https://data.cami-challenge.org/">CAMI portal</a>. You can download this file in the section “2nd CAMI Challenge Marine Dataset”</li>
      <li>
        <p>You can download data with CAMI client using the following command line in the terminal:</p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>java -jar camiClient.jar -d &lt;linkfileLocation&gt; &lt;targetDirectory&gt; -p PATTERN
</code></pre></div>        </div>
      </li>
    </ol>
  </blockquote>

</blockquote>

<h4 id="upload-data-into-galaxy">Upload data into <span class="notranslate">Galaxy</span></h4>

<p>There are different ways to upload data into the <span class="notranslate">Galaxy</span>. In case of a not very large dataset it’s more convenient to upload data directly from your computer to <span class="notranslate">Galaxy</span>. In this tutorial we use a large dataset, thus, we use FTP server for this purpose.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-upload-data-into-galaxy-with-ftp"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Upload data into <span class="notranslate">Galaxy</span> with FTP</h3>

  <ol>
    <li>
      <p>Create new history</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>
      <p>Upload datasets into FTP <span class="notranslate">Galaxy</span> server</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-upload-many-files-10-via-ftp" class="box-title"><button type="button" aria-controls="tip-upload-many-files-10-via-ftp-contents" aria-expanded="true" aria-label="Toggle tip box: Upload many files (&gt;10) via FTP"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload many files (&gt;10) via FTP<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <ol>  <li>    <p>Make sure to have an FTP client installed</p>    <p>There are many options. We can recommend <a href="https://filezilla-project.org/">FileZilla</a>, a free FTP client that is available on Windows, MacOS, and Linux.</p>  </li>  <li>Establish FTP connection to the <span class="notranslate">Galaxy</span> server    <ol>      <li>Provide the <span class="notranslate">Galaxy</span> server’s FTP server name (e.g. <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.org</code>, <code class="language-plaintext highlighter-rouge">ftp.use<span class="notranslate">galaxy</span>.eu</code>)</li>      <li>Provide the <strong>use<span class="notranslate">rna</span>me</strong> (usually the email address) and the <strong>password</strong> on the <span class="notranslate">Galaxy</span> server</li>      <li>Connect</li>    </ol>  </li>  <li>    <p>Add the files to the FTP server by dragging/dropping them or right clicking on them and uploading them</p>    <p>The FTP transfer will start. We need to wait until they are done.</p>  </li>  <li>Open the Upload menu on the <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Choose FTP file</strong> on the bottom</li>  <li>Select files to import into the history</li>  <li>Click on <strong>Start</strong></li></ol></blockquote>
    </li>
    <li>
      <p>Press “upload data” -&gt; “choose remote files” -&gt; “FTP directory” and upload your data to your history from FTP server</p>
    </li>
  </ol>
</blockquote>

<h4 id="preprocessing-data">Preprocessing data</h4>

<p>We got the marine data in our history, with one file per sample (10 samples for long <span class="notranslate">reads</span> and 10 samples for short <span class="notranslate">reads</span>). We need to prepare our data.</p>

<ol>
  <li>We should organise our data into collections. For deeper learning about dataset collections follow <span class="citation"><a href="#noauthor_galaxy_nodate-1">“<b><span class="notranslate">Galaxy</span> Training: Using dataset collections</b>”</a></span> tutorial.</li>
  <li>These files consist of interleaved paired-end <span class="notranslate">reads</span>, i.e. forward and reverse <span class="notranslate">reads</span> are all together in one single file and not 2 as often. This can be a problem for some assemblers as not every assembling tool can take interleaved paired-end <span class="notranslate">reads</span> as input.For co-assembly of all samples, some assemblers expect one file with all samples  combined together.</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-preprocessing-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Preprocessing data</h3>

  <ol>
    <li>Take files for short <span class="notranslate">reads</span> for all samples</li>
    <li>Create a collection with all samples- We now have a collection with all files. Let’s now combine all sample files into one file</li>
    <li>
      <p>Combine all samples from collection into one file -  follow the section “<a href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html#collapse-data-into-a-single-dataset">Collapse data into a single dataset</a>” in <span class="citation"><a href="#noauthor_galaxy_nodate-1">“<b><span class="notranslate">Galaxy</span> Training: Using dataset collections</b>”</a></span></p>

      <p>We can now split forward and reverse <span class="notranslate">reads</span> into 2 files</p>
    </li>
    <li><strong>FASTQ splitter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ <span class="notranslate">reads</span>”</em>: output of Collapse collection tool</li>
      </ul>
    </li>
    <li>Check if the names of the output generated by Fastq splitter include the information about forward and reverse, rename them otherwise</li>
    <li>Repeat 1-5 steps for long <span class="notranslate">reads</span>. Eventually we have 2 collections (one for short <span class="notranslate">reads</span>, one for long <span class="notranslate">reads</span>)</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h5 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h5>
  <p>While working on this project we realised that both outputs (forward and reverse) had the same name - “FASTQ splitter on data X”, where X - is the inte<span class="notranslate">rna</span>l number of data in this <span class="notranslate">Galaxy</span> history. Since we have run FASTQ splitter on one dataset and got 2 outputs, as a result filenames of both outputs were the same. Why is it not good? It can be mistaken by assembler (or any other tool where we want to use FASTQ splitter output files) because it will not know which file is reverse and which is forward.We updated the <span class="notranslate">Galaxy</span> tool to fix this issue</p>
</blockquote>

<h2 id="select-and-prepare-the-tools-to-run">Select and prepare the tools to run</h2>

<p>Based on <a href="https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1.full">CAMI2 paper</a> [2] and <a href="https://www.nature.com/articles/s41596-020-00480-3#Bib1">CAMI2 tutorial paper</a> [20], we can compare tools on a set of metrics to select the one to use for an analysis but also here to run the challenge. Tools performance ranking.</p>

<p><strong><em>Performance of assembly tools on the marine dataset</em></strong>:</p>
<iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=1873156403&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

<blockquote class="details">
  <h3 id="details-performance-of-the-assembly-tools-on-the-strain-madness-dataset"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Performance of the assembly tools on the strain madness dataset</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=153898462&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<blockquote class="details">
  <h3 id="details-performance-of-the-assembly-tools-on-the-plant-associated-dataset"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Performance of the assembly tools on the plant-associated dataset</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=694755016&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>Using the described metrics, the different tools were evaluated in the CAMI paper and aggregated in tables (Supplementary Tables 3-7) from CAMI tutorial [20].</p>

<p>In these tables there are also ranking scores of the tools shown for every statistic as well as overall ranking scores. Overall, ranking scores for every dataset are computed as a sum of all ranking scores across metrics. The average ranking score of both datasets are calculated as weighted average sum of ranking for both datasets. We created a table showing all ranking results from previous tables:</p>

<blockquote class="details">
  <h3 id="details-assembly-tools-ranking"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Assembly tools ranking</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=455354696&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>We summed up ranking scores for all methods, common genomes, and unique genomes to extract the following  priority list for tools</p>

<p><strong>Marine</strong> dataset</p>

<p><em>With tool versions</em></p>
<ol>
  <li>HipMer</li>
  <li>metaSPAdes_v3.13.1</li>
  <li>metaSPAdes_v3.13.0</li>
  <li>ABySS</li>
  <li>Ray-Meta</li>
  <li>Megahit_v1.1.2</li>
  <li>SPAdes_v3.14-dev</li>
</ol>

<p><em>Without tool versions</em></p>
<ol>
  <li>HipMer</li>
  <li><strong>metaSPAdes</strong></li>
  <li><strong>ABySS</strong></li>
  <li><strong>Ray-Meta</strong></li>
  <li><strong>Megahit</strong></li>
</ol>

<p><strong>Strain madness</strong> dataset
<em>With tool versions</em></p>
<ol>
  <li>HipMer</li>
  <li>Megahit_v1.1.2</li>
  <li>SPAdes_v3.14-dev</li>
  <li>OPERA-MS</li>
  <li>Megahit_V1.2.7</li>
</ol>

<p><em>Without tool versions</em></p>
<ol>
  <li>HipMer</li>
  <li><strong>Megahit</strong></li>
  <li><strong>SPAdes</strong></li>
  <li><strong>OPERA</strong></li>
</ol>

<p><strong>Plant-associated</strong> dataset</p>

<p>There are no certain ranking tables among Supplementary tables for plant-associated dataset. However,  in the <a href="https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1.full">paper</a> [2] there is information related to tools performance on plant-associated dataset. We created the priority list of tools for the plant-associated dataset.</p>

<ol>
  <li>(Meta)HipMer</li>
  <li><strong>(meta)Flye</strong></li>
  <li><strong>(meta)SPAdes</strong></li>
</ol>

<p>Since in this tutorial we have decided to focus on marine dataset it would be reasonable to reproduce CAMI2 assembly challenge using HipMer, metaSPAdes, ABySS, Ray-Meta, Megahit assemblers which performed better. As we know from our comparison <span class="notranslate">Galaxy</span> and CAMI2 analysis, metaSPAdes, ABySS, Megahit tools are available in <span class="notranslate">Galaxy</span> while Ray-Meta and HipMer are not.</p>

<blockquote class="details">
  <h3 id="details-assembly-tools-comparison"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Assembly tools comparison</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=0&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>Moreover, we also selected Flye assembler, even though Flye is not in the Top-5 assemblers in CAMI2. Flye is a long-read assembler that works well for this task.. We do that to show the assembler launched on long-<span class="notranslate">reads</span> and not only on short-<span class="notranslate">reads</span> and hybrid.</p>

<h3 id="tool-update">Tool Update</h3>

<p>Due to the limited time for this project, we have decided to skip adding new tools into <span class="notranslate">Galaxy</span>. So, in this tutorial we do not run Hipmer and Ray-Meta.</p>

<p>But, in order to show the possibility of adding, updating and fixing tools and the process of how to do that.</p>

<p>All tools in <span class="notranslate">Galaxy</span> are kept up-to-date with their corresponding command-line tool. This is why in this tutorial we checked if the tool we want to use is up-to-date before using it. We then updated Megahit from 1.1.3.5 version to 1.2.9 (the latest version for that moment).</p>

<p>The other case it might be necessary to update the tool when we get an error on tool run or we find out that there is a bug. It can be an issue/bug in the tool itself, then we have to contact the tool developers. In other cases, it can be an issue/bug with the tool wrapper. If this is the case, you may be able to fix it by yourself.</p>

<p>This was the case with the metaSPAdes tool. When we launched the metaSPAdes tool it failed with the error:</p>

<p>ln: failed to create symbolic link ‘paired_<span class="notranslate">reads</span>1/FASTQ_splitter_on_data_X.fastq’: File exists</p>

<p>The reason for this error was the same filenames with forward and reverse <span class="notranslate">reads</span>. We figure out that this error came from the FASTQ splitter tool, so we updated the FASTQ splitter. But we also updated spades tool to avoid similar cases in the future.</p>

<p>Updating or fixing  tools has the same concept as adding tools but is less time consuming. Therefore, in this tutorial we will limit ourselves to fixing a tool, but the concept will be the same for adding or updating a new tool.</p>

<p>Now we are going to update the tool in <span class="notranslate">Galaxy</span> by updating its wrapper. A wrapper is a XML which serves as a layer between the command-line tool itself and the <span class="notranslate">Galaxy</span> interface. This wrapper helps <span class="notranslate">Galaxy</span> users to have a visual interface and selection of the parameters of the command-line.  Please read our dedicated tutorial to learn more about <span class="notranslate">Galaxy</span> tools and the detailed instruction of how to build or update the tool is in <span class="citation"><a href="#noauthor_building_nodate">“<b>Building <span class="notranslate">Galaxy</span> Tools Using Planemo — Planemo 0.75.0.dev0 documentation</b>”</a></span>.</p>

<p>To update or fix a tool, we need to change its XML. Tools in a <span class="notranslate">Galaxy</span> server are installed by the administrators of this <span class="notranslate">Galaxy</span> server using the <span class="notranslate">Galaxy</span> <span class="notranslate">ToolShed</span>. The <span class="notranslate">Galaxy</span> <span class="notranslate">ToolShed</span> is like an AppStore with the XML for all possible <span class="notranslate">Galaxy</span> tools. So to update a tool, we need to change its XML in the <span class="notranslate">ToolShed</span>. 
But usually, XML of tools are developed collaboratively inside GitHub repositories and then pushed to the <span class="notranslate">ToolShed</span>. So we need to find the development repository of the <span class="notranslate">Galaxy</span> tool to update it.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-finding-the-development-repository-of-a-tool-to-update"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Finding the development repository of a tool to update</h3>

  <ol>
    <li>Open the fastqc splitter in <span class="notranslate">Galaxy</span></li>
    <li>Click on “See in the <span class="notranslate">ToolShed</span>” on the top right of the tool form. You will be redirected to the tool on the <span class="notranslate">ToolShed</span></li>
    <li>Follow the link in “<strong>Development repository”</strong></li>
    <li><strong>You have now the tool wrapper, i.e. its XML</strong></li>
    <li>Once we have the development repository setup, we can update the tool. We recommend using Planemo, the <span class="notranslate">Galaxy</span> software development kit, to help the process. More information is in <span class="citation"><a href="#noauthor_building_nodate">“<b>Building <span class="notranslate">Galaxy</span> Tools Using Planemo — Planemo 0.75.0.dev0 documentation</b>”</a></span>.</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-update-the-tools"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Update the tools</h3>

  <ol>
    <li>Update lines corresponding to output files: add “Forward” or “Reverse” to the filenames respectively.</li>
    <li>Update tests for this tool</li>
    <li>Lint and test the tool using <code class="language-plaintext highlighter-rouge">planemo lint</code> and <code class="language-plaintext highlighter-rouge">planemo test</code></li>
    <li>Push the changes to a branch of your fork of the GitHub repository</li>
    <li>Create a Pull Request</li>
    <li>Modify the tool given the suggested changes</li>
    <li>In case you add new tool - add the tool to yaml file (for use<span class="notranslate">galaxy</span>.eu server - <a href="https://github.com/usegalaxy-eu/usegalaxy-eu-tools/blob/master/tools_iuc.yaml">tools_iuc.yaml</a>)</li>
    <li>Once the Pull request is merged to the main branch at GitHub, the tool will be added (or updated with increased version) in Tool Shed (for use<span class="notranslate">galaxy</span>.eu - <a href="https://toolshed.g2.bx.psu.edu/"><span class="notranslate">Galaxy</span> - Tool Shed</a>) on weekend</li>
    <li>Go to the Tool Shed after weekend and check the tool changes</li>
    <li>Go to use<span class="notranslate">galaxy</span>.eu (or to other <span class="notranslate">Galaxy</span> server if you used one) and check that tool was added/updated</li>
  </ol>
</blockquote>

<h3 id="run-the-tools">Run the tools</h3>

<p>We run tools both on individual sample and on all pooled samples as it was done in CAMI2 challenge as well.</p>

<p>It is important to use both approaches: individual assembly and co-assembly and benchmark them in order to get the intuition of more applicable technique for the certain dataset. In case of individual assembly we have the possibility to de-replicate genome after binning. It will not be done in this tutorial as the purpose here is only assembly CAMI2 challenge. However, you can get to know both these practices in the <span class="citation"><a href="#noauthor_galaxy_nodate-4">“<b><span class="notranslate">Galaxy</span> Training: Assembly</b>”</a></span> (in the section about de-replication).</p>

<p>Co-assembly is more commonly used than individual assembly and then de-replication after binning. Nevertheless, de-replication approach should be taken into account while choosing the assembly method.</p>

<p>The runtime and memory usage are obviously different for not only these two scenarios, but also for the tools. Some assemblers are fast (like Megahit), some are slow (like Flye).</p>

<p>While choosing parameters to run every tool with we oriented on parameters which have been used in CAMI2 challenge by its participants. However, we didn’t consider it as absolute truth as in metagenomic analysis it can be not very clear which tools and which parameters would be better for the best assembly performance. Then we had to be critical to what had been done in CAMI2. It was the basis of this tutorial/project, even though we made some corrections. Additionally, we take into account recommendations from tool documentation.</p>

<p>A <span class="notranslate">Galaxy</span> history was created for each tool and type of assembly (single or co-assembly). We named them such that it is easy to identify the tool and type from the mame, but we also Created a sp<span class="notranslate">reads</span>heet to list all histories along with their metadata (tool, etc).</p>

<h4 id="long-reads">Long <span class="notranslate">reads</span></h4>

<p><strong>MetaFlye</strong> (metagenome mode of Flye) is a scalable long-read metagenome assembly using repeat graphs [<a href="https://www.biorxiv.org/lookup/google-scholar?link_type=googlescholar&amp;gs_type=article&amp;author[0]=M.+Kolmogorov&amp;title=metaFlye:+scalable+long-read+metagenome+assembly+using+repeat+graphs&amp;publication_year=2020&amp;journal=Nat.+Methods&amp;volume=17&amp;pages=1103-1110">26</a>]. It is supposed to use long <span class="notranslate">reads</span> as input.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-co-assembly-of-long-read-samples-with-flye"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Co-assembly of long-read samples with Flye</h3>

  <ol>
    <li>
      <p>Create a new history with the name “CAMI2 Flye v29 -meta -nano-raw pooled interleaved long collection”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-1" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-1-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>
      <p>Run <strong>Flye</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:</p>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input <span class="notranslate">reads</span>”</em>: the collection of long <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mode”</em>:	–nano-raw</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Perform metagenomic assembly”</em>: True</li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>Flye was reported in CAMI2 using parameter Mode –-raw-bio while datasets were sequenced with nanopore, thus we corrected parameters and used –-nano-raw instead.</p>
  </blockquote>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-individual-assembly-of-long-read-sample-with-flye"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Individual assembly of long-read sample with Flye</h3>

  <ol>
    <li>
      <p>Create a new history with the name “CAMI2 Flye v29 -meta -nano-raw pooled interleaved long collection”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-2" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-2-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>
      <p>Run <strong>Flye</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:</p>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input <span class="notranslate">reads</span>”</em>: long <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mode”</em>:	–nano-raw</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Perform metagenomic assembly”</em>: True</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>We just launched Flye, first, for collection of long <span class="notranslate">reads</span> from all 10 samples to produce co-assembly. In the second Hands-on we launched Flye on individual sample. In the following hands-on for other tools for short-<span class="notranslate">reads</span> and hybrid we will need to run assemblers for both options as well.</p>

<h4 id="short-reads">Short <span class="notranslate">reads</span></h4>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-co-assembly-of-short-reads-samples-with-megahit-abyss-and-metaspades"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Co-assembly of short-<span class="notranslate">reads</span> samples with Megahit, Abyss, and MetaSPAdes</h3>

  <ol>
    <li>
      <p>Create a new history with the name “Co-assembly of short-<span class="notranslate">reads</span> samples with Megahit, Abyss, and MetaSPAdes”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-3" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-3-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>In the next steps we use the <strong>Megahit</strong> assembler. MEGAHIT is an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph [10]. As input it uses metagenomics samples as paired-end fastq files. As metagenomics input data short <span class="notranslate">reads</span> should be used.</p>
  </blockquote>

  <ol>
    <li>Run <strong>MEGAHIT</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select your input option”</em>: interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved-paired-end file(s)”</em>: the collection of short <span class="notranslate">reads</span> from all samples</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MEGAHIT v129 pooled interleaved short collection”</li>
    <li>Run <strong>MEGAHIT</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select your input option”</em>: paired</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mate 1 input <span class="notranslate">reads</span>”</em>: Output of FASTQ splitter: Forward</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mate 2 input <span class="notranslate">reads</span>”</em>: Output of FASTQ splitter: Reverse</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer specification method”</em>: klim_method</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Minimum kmer size”</em>: 21</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Maximum kmer size”</em>: 91</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Increment of kmer size of each iteration”</em>: 12</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MEGAHIT v129 paired-end k-min21 k-max91 k-step12 pooled deinterleaved short collapsed”</li>
    <li>Run <strong>MEGAHIT</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select your input option”</em>: interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved-paired-end file(s)”</em>: Output of FASTQ splitter: the collection of short <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer specification method”</em>: klim_method</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Minimum kmer size”</em>: 21</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Maximum kmer size”</em>: 91</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Increment of kmer size of each iteration”</em>: 10</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MEGAHIT v129 k-min21 k-max91 k-step10 pooled interleaved short collection”</li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>In the next steps we use <strong>Abyss</strong> assembler. ABySS is a parallel assembler for short read sequence data [23]. It uses de novo short read assembly algorithms. Short <span class="notranslate">reads</span> should be used as input data for Abyss.</p>
  </blockquote>

  <ol>
    <li>Run <strong>ABYSS</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of paired-end datasets”</em>: paired_il</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved paired-end <span class="notranslate">reads</span>”</em>: collapsed collection of short <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer length (in bp)”</em>: 41</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 Abyss v234 k41 pooled interleaved short collapsed”</li>
    <li>Run <strong>ABYSS</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of paired-end datasets”</em>: paired_il</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved paired-end <span class="notranslate">reads</span>”</em>: collapsed collection of short <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer length (in bp)”</em>: 96</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 Abyss v234 k96 pooled interleaved short collapsed”</li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>In the next steps we use <strong>metaSPAdes</strong> assembler. MetaSPAdes is a versatile metagenomic assembler [9]. As input for metaspades it can accept short <span class="notranslate">reads</span>. However, there is an option to use additionally long <span class="notranslate">reads</span> besides short <span class="notranslate">reads</span> to produce hybrid input.</p>
  </blockquote>

  <ol>
    <li>Run <strong>MetaSPAdes</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Pair-end <span class="notranslate">reads</span> input format”</em>: paired_interlaced</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ file(s): interlaced”</em>: collapsed collection of short <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select k-mer detection option”</em>: manual</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer size values”</em>: 21,33,55,77</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MetaSPAdes v3_15_3 k21-33-55-77 pooled interleaved short collapsed”</li>
    <li>Run <strong>MetaSPAdes</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Pair-end <span class="notranslate">reads</span> input format”</em>: paired_interlaced</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ file(s): interlaced”</em>: collapsed collection of short <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Arf - Nanopore <span class="notranslate">reads</span>”</em>: collapsed collection of long <span class="notranslate">reads</span> from all samples</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select k-mer detection option”</em>: manual</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer size values”</em>: 21,33,55,77</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MetaSPAdes v3_15_3 k21-33-55-77 pooled interleaved short collapsed additional nanopore long collapsed”</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-individual-assembly-of-short-reads-samples-with-megahit-abyss-and-metaspades"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Individual assembly of short-<span class="notranslate">reads</span> samples with Megahit, Abyss, and MetaSPAdes</h3>

  <ol>
    <li>
      <p>Create a new history with the name “Individual of short-<span class="notranslate">reads</span> samples with Megahit, Abyss, and MetaSPAdes”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-4" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-4-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>In the next steps we use the <strong>Megahit</strong> assembler. MEGAHIT is an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph [10]. As input it uses metagenomics samples as paired-end fastq files. As metagenomics input data short <span class="notranslate">reads</span> should be used.</p>
  </blockquote>

  <ol>
    <li>Run <strong>MEGAHIT</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select your input option”</em>: interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved-paired-end file(s)”</em>: dataset of short <span class="notranslate">reads</span> from sample1</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MEGAHIT v129 interleaved short sample1”</li>
    <li>Run <strong>MEGAHIT</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select your input option”</em>: paired</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mate 1 input <span class="notranslate">reads</span>”</em>: Output of FASTQ splitter: Forward</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mate 2 input <span class="notranslate">reads</span>”</em>: Output of FASTQ splitter: Reverse</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer specification method”</em>: klim_method</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Minimum kmer size”</em>: 21</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Maximum kmer size”</em>: 91</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Increment of kmer size of each iteration”</em>: 12</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MEGAHIT v129 paired-end k-min21 k-max91 k-step12 deinterleaved short sample1”</li>
    <li>Run <strong>MEGAHIT</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select your input option”</em>: interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved-paired-end file(s)”</em>: Output of FASTQ splitter: dataset of short <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer specification method”</em>: klim_method</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Minimum kmer size”</em>: 21</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Maximum kmer size”</em>: 91</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Increment of kmer size of each iteration”</em>: 10</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MEGAHIT v129 k-min21 k-max91 k-step10 interleaved short sample1”</li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-6"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>In the next steps we use <strong>Abyss</strong> assembler. ABySS is a parallel assembler for short read sequence data [23]. It uses de novo short read assembly algorithms. Short <span class="notranslate">reads</span> should be used as input data for Abyss.</p>
  </blockquote>

  <ol>
    <li>Run <strong>ABYSS</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of paired-end datasets”</em>: paired_il</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved paired-end <span class="notranslate">reads</span>”</em>: dataset of short <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer length (in bp)”</em>: 41</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 Abyss v234 k41 interleaved short sample1”</li>
    <li>Run <strong>ABYSS</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of paired-end datasets”</em>: paired_il</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved paired-end <span class="notranslate">reads</span>”</em>: dataset of short <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer length (in bp)”</em>: 96</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 Abyss v234 k96 interleaved short sample1”</li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-7"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>In the next steps we use <strong>metaSPAdes</strong> assembler. MetaSPAdes is a versatile metagenomic assembler [9]. As input for metaspades it can accept short <span class="notranslate">reads</span>. However, there is an option to use additionally long <span class="notranslate">reads</span> besides short <span class="notranslate">reads</span> to produce hybrid input.</p>
  </blockquote>

  <ol>
    <li>Run <strong>MetaSPAdes</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Pair-end <span class="notranslate">reads</span> input format”</em>: paired_interlaced</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ file(s): interlaced”</em>: dataset of short <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select k-mer detection option”</em>: manual</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer size values”</em>: 21,33,55,77</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MetaSPAdes v3_15_3 k21-33-55-77 interleaved short sample1”</li>
    <li>Run <strong>MetaSPAdes</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Pair-end <span class="notranslate">reads</span> input format”</em>: paired_interlaced</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ file(s): interlaced”</em>: dataset of short <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Arf - Nanopore <span class="notranslate">reads</span>”</em>: dataset of long <span class="notranslate">reads</span> from sample1</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select k-mer detection option”</em>: manual</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“K-mer size values”</em>: 21,33,55,77</li>
      </ul>
    </li>
    <li>Rename output to “Output - CAMI2 MetaSPAdes v3_15_3 k21-33-55-77 interleaved short sample1 additional nanopore long sample1”</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-8"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>We got an error “Out of memory” for the first launches of metaspades. The memory limit in <span class="notranslate">Galaxy</span> was set to 200 Gb. After that,<a href="https://github.com/usegalaxy-eu/infrastructure-playbook/pull/358"> the memory for this tool was increased up to 250 GB</a>, which is the recommended default value according to <a href="http://cab.spbu.ru/files/release3.12.0/manual.html#sec3.2">SPAdes documentation</a> (see “Advanced options” section). Only <span class="notranslate">Galaxy</span> administrators can solve this kind of problem with memory. If you need the memory increase for the specific tool you need to report an issue via <span class="notranslate">Galaxy</span> by clicking “View and report this error” (small “bug” icon).</p>

  <p>However, it was not enough. According to <a href="https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1.supplementary-material">Supplementary Table 2</a> CAMI2 participants used the memory limit of 500 Gb. Since the memory limit in <span class="notranslate">Galaxy</span> for every tool is set once and it is used for every launch of this tool no matter how much memory it is needed. It seems to be unreasonable to set 500 Gb instead of the recommended 250, because then any spades job would request that amount of memory.</p>

  <p>Other issues we encountered during this project are more general. Protracted technical works occured on the <span class="notranslate">Galaxy</span> platform, all tools were stopped and launched from scratch. That means we ran out of time. Before these technical works tools have been running for around one month and they were re-launched. However, we increased the CPU which helped us to get results sooner.</p>
</blockquote>

<h3 id="description-of-metrics">Description of metrics</h3>

<p>Several metrics are used to assess the quality of the assemblies.</p>

<ul>
  <li><strong>Genome fraction (%)</strong>: % of reference bases covered by assembled contigs obtained by similarity-based <span class="notranslate">mapping</span>.
    <ul>
      <li>The total number of aligned bases in the reference, divided by the genome size. A base in the reference genome is counted as aligned if at least one contig has at least one alignment to this base. Contigs from repeat regions may map to multiple places, and thus may be counted multiple times in this quantity.</li>
    </ul>
  </li>
  <li><strong>Mismatches per 100 kbp</strong>: number of mismatched bases in the contig-reference alignment (average per 100 kb).
    <ul>
      <li>The average number of mismatches per 100 000 aligned bases. This metric does not distinguish between <em>single-nucleotide polymorphisms,</em> which are true differences in the assembled genome versus the reference genome, and <em>single-nucleotide errors,</em> which are due to errors in <span class="notranslate">reads</span> or errors in the assembly algorithm.</li>
      <li><strong>Duplication ratio</strong>: total number of aligned bases / genome fraction * reference length.</li>
    </ul>
  </li>
  <li><strong>Number of misassemblies</strong>: number of contigs which:</li>
  <li>contain a gap of more than 1kb;</li>
  <li>contain inserts of more than 1kb; or</li>
  <li>align to different genomes</li>
  <li><strong>The number of misassemblies</strong>: number of positions in the assembled contigs where the left flanking sequence aligns over 1 kb away from the right flanking sequence on the reference, or they overlap by &gt;1 kb, or the flanking sequences align on opposite strands or different chromosomes.</li>
  <li><strong>NGA50</strong>:  metric for measuring the contiguity of an assembly.
    <ul>
      <li>For each reference genome, all contigs aligned to it are sorted by size and the NGA50 for that genome is defined as the length of the contig cumulatively surpassing 50% genome fraction. If a genome is not covered to 50%, NGA50 is undefined. Since we report the average NGA50 over all genomes, it was set to 0 for genomes with less than 50% genome fraction.</li>
      <li>There is a similar metric NG50 that defines the contig length such that using equal or longer length contigs produces 50% of the length of the reference genome. NGA50 is NG50 such that the lengths of aligned blocks are counted instead of contig lengths. It means that first we perform the sequence alignment of contigs to the reference genome. Next, contigs with misassemblies are split into aligned blocks and are aligned independently to distinct parts of the genome.</li>
    </ul>
  </li>
  <li><strong>Strain recall</strong>: raction of high-quality (more than 90% genome fraction and less than 100 mismatches per 100 kb) genome assemblies recovered for all ground truth genomes. Strain recall measures how many genomes are recovered with high genome fraction and few mismatches (mm).</li>
  <li><strong>Strain precision</strong>: fraction of high-quality assemblies among all high genome fraction (more than 90%) assemblies.
    <ul>
      <li>Strain precision assesses how accurately reference genomes are recovered, based on the fraction of correctly assembled high-quality, near-complete genomes (&gt;90% genome fraction, &lt;0.1% mm) divided by the overall number of assembled, near-complete genomes (&gt;90% genome fraction).</li>
    </ul>
  </li>
</ul>

<h2 id="assess-the-assembly-outputs">Assess the assembly outputs</h2>

<p>In CAMI2, different tools are used to evaluate the quality of the tools for the different challenges.</p>

<blockquote class="details">
  <h3 id="details-the-list-of-assessment-tools-with-metadata"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> The list of assessment tools with metadata</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=15494775&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>In this tutorial we focus on the assembly challenge and on the metrics presented before.</p>

<p>Before running Quast as in CAMI to extract general statistics, we want to know the percentage of <span class="notranslate">reads</span> that were used to build the assemblie</p>

<h3 id="extraction-of--reads-used-in-assemblies">Extraction of % <span class="notranslate">reads</span> used in assemblies</h3>

<p>To extract this information, we map the input <span class="notranslate">reads</span> on the assemblies.</p>

<h4 id="short-reads-1">Short-<span class="notranslate">reads</span></h4>

<p>We use Bowtie2 for <span class="notranslate">mapping</span> short-<span class="notranslate">reads</span> raw data to the assembly we got after usage of different short-<span class="notranslate">reads</span> assemblers (Megahit, MetaSPAdes, Abyss) in order to get the percentage of <span class="notranslate">reads</span> used for assembly.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-percentage-of-reads-used-for-co-assembly-with-bowtie2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get percentage of <span class="notranslate">reads</span> used for co-assembly with Bowtie2</h3>

  <ol>
    <li>
      <p>Create a new history named “Short <span class="notranslate">reads</span> co-assemblies”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-5" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-5-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Drag and drop /results from the different reshistories with short read co-assemblies</li>
    <li>Run <strong>Bowtie2</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Is this single or paired library”</em>: paired_interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved FASTQ file”</em>: raw dataset collection</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Will you select a reference genome from your history or use a built-in index”</em>: history</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select reference genome”</em>: Megahit/MetaSPAdes/Abyss output (If the assembler, like Abyss, has Contigs and Scaffolds as output, use Contigs preferably. Contigs do not contain gaps represented with multiple-X letters.)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Save the bowtie2 <span class="notranslate">mapping</span> statistics to the history”</em>: True</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-percentage-of-reads-used-for-individual-assembly-with-bowtie2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get percentage of <span class="notranslate">reads</span> used for individual assembly with Bowtie2</h3>

  <ol>
    <li>
      <p>Create a new history named “Short <span class="notranslate">reads</span> co-assemblies”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-6" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-6-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Drag and drop /results from the different reshistories with short read individual assemblies</li>
    <li>Run <strong>Bowtie2</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Is this single or paired library”</em>: paired_interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Interleaved FASTQ file”</em>: raw dataset collection</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Will you select a reference genome from your history or use a built-in index”</em>: history</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select reference genome”</em>: Megahit/MetaSPAdes/Abyss output (If the assembler, like Abyss, has Contigs and Scaffolds as output, use Contigs preferably. Contigs do not contain gaps represented with multiple-X letters.)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Save the bowtie2 <span class="notranslate">mapping</span> statistics to the history”</em>: True</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<h4 id="long-reads-1">Long <span class="notranslate">reads</span></h4>

<p>Bowtie2 should not be used to align long <span class="notranslate">reads</span> to resulting assemblies because, as it is stated in Bowtie2 documentation, “Bowtie2 is geared toward aligning relatively short sequencing <span class="notranslate">reads</span> to long genomes. That said, it handles arbitrarily small reference sequences (e.g. amplicons) and very long <span class="notranslate">reads</span> (i.e. upwards of 10s or 100s of kilobases), though it is slower in those settings. It is optimised for the read lengths and error modes yielded by typical Illumina sequencers”.</p>

<p>For long-<span class="notranslate">reads</span> instead, <em>Minimap2</em> aligner can be used for <span class="notranslate">mapping</span> the Nanopore-sequenced data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-percentage-of-reads-used-for-co-assembly-with-map-with-minimap2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get percentage of <span class="notranslate">reads</span> used for co-assembly with Map with minimap2</h3>

  <ol>
    <li>
      <p>Create a new history named “Long <span class="notranslate">reads</span> co-assemblies”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-7" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-7-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Drag and drop /results from the different reshistories with short read co-assemblies</li>
    <li>Run <strong>Map with minimap2</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Will you select a reference genome from your history or use a built-in index?”</em>: Use a genome from history and build index</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: output of Flye</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Single or Paired-end <span class="notranslate">reads</span>”</em>: paired interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: raw dataset collection
This tool run produces one collection with the actual mapped <span class="notranslate">reads</span> for each Nanopore-sequenced sample. Unlike Bowtie2 it does not have an option to output <span class="notranslate">mapping</span> statistics directly. However, we can generate that information through an extra step.</li>
      </ul>
    </li>
    <li>Run <strong>Samtools stat</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM file”</em>: the collection of mapped Nanopore-sequenced <span class="notranslate">reads</span>, output of Map with minimap2 tool</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output”</em>: One single summary file</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-percentage-of-reads-used-for-individual-assembly-with-map-with-minimap2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get percentage of <span class="notranslate">reads</span> used for individual assembly with Map with minimap2</h3>

  <ol>
    <li>
      <p>Create a new history named “Long <span class="notranslate">reads</span> individual assemblies”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-8" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-8-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Drag and drop /results from the different reshistories with short read co-assemblies</li>
    <li>Run <strong>Map with minimap2</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Will you select a reference genome from your history or use a built-in index?”</em>: Use a genome from history and build index</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: output of Flye</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Single or Paired-end <span class="notranslate">reads</span>”</em>: paired interleaved</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: raw dataset collection</li>
      </ul>
    </li>
    <li>Run <strong>Samtools stat</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM file”</em>: the collection of mapped Nanopore-sequenced <span class="notranslate">reads</span>, output of Map with minimap2 tool</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output”</em>: One single summary file</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<h3 id="extraction-of-general-metrics">Extraction of general metrics</h3>

<p>Assemblies were evaluated with metaQUAST (metagenomics mode of QUAST) version 5.0.2.</p>

<h4 id="quast">Quast</h4>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-metaquast-without-provided-reference-genome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Metaquast without provided reference genome</h3>

  <ol>
    <li>Move to “Short <span class="notranslate">reads</span> co-assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Move to “Short <span class="notranslate">reads</span> individual assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Move to “Long <span class="notranslate">reads</span> co-assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Move to “Long <span class="notranslate">reads</span> individual assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-metaquast-with-provided-reference-genome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Metaquast with provided reference genome</h3>

  <ol>
    <li>Move to “Short <span class="notranslate">reads</span> co-assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: GSA file from CAMI2 data</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-comment-9"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>Working on this tutorial we encountered some issues. For example, in CAMI2 Quast was launched with Gold Standard Assembly as a reference genome. In <span class="notranslate">Galaxy</span> it took a long time and the tool is still running. Withsout GSA as a reference genome we got QUAST results faster.</p>
  </blockquote>

  <ol>
    <li>Move to “Short <span class="notranslate">reads</span> individual assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: GSA file from CAMI2 data</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Move to “Long <span class="notranslate">reads</span> co-assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: GSA file from CAMI2 data</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Move to “Long <span class="notranslate">reads</span> individual assemblies” history</li>
    <li>Run <strong>Quast</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: outputs of the assemblies</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Type of assembly”</em>: metagenome</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: GSA file from CAMI2 data</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Output files”</em>: HTML reports, PDF reports, Tabular reports, Log file</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-10"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>While running Quast, some of the generated outputs (statistics and log files) were empty while the html report was not. After investigation, it seems that the outputs were not correctly retrieved in the <span class="notranslate">Galaxy</span> wrapper. We then fixed the tool.</p>
</blockquote>

<h4 id="aggregate-all-metrics">Aggregate all metrics</h4>

<p>To generate a nice report with all metrics combined we use MultiQC. This tool is a good choice when you want to combine results and make them visually represented with different graphs. MultiQC generates a webpage combining reports.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-aggregate-all-metrics-with-multiqc"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Aggregate all metrics with MultiQC</h3>

  <ol>
    <li>
      <p>Create a new history named “Benchmarking analysis of reproduced CAMI2”</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <div id="tip-creating-a-new-history-9" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-9-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Drag&amp;drop Quast, bowtie2 and Samtools stat outputs from all histories for different assemblers</li>
    <li>Run <strong>MultiQC</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Which tool used to generate report”</em>: Quast</li>
      </ul>
    </li>
    <li>(for long <span class="notranslate">reads</span>) <strong>Add report</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Which tool used to generate report”</em>: samtools</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Data”</em>: Output of Samtool stats&gt;</li>
      </ul>
    </li>
    <li>(for short <span class="notranslate">reads</span>) <strong>Add report</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Which tool used to generate report”</em>: Bowtie 2</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Data”</em>: all assemblers outputs of Bowtie2 (for short <span class="notranslate">reads</span>)</li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<h3 id="extract-computational-metrics-from-galaxy---memory-used-runtime">Extract computational metrics from <span class="notranslate">Galaxy</span> - memory used, runtime</h3>

<p>In CAMI2 there were compared computational characteristics of different tools such as runtime and memory usage. There were created the following plots:</p>

<p><img src="../../images/cami2/tools-job-statistics.jpg" alt="Job statistics from CAMI2. " width="70%" loading="lazy" /></p>

<p>To export job metrics (memory usage and runtime) from <span class="notranslate">Galaxy</span> in csv format use <a href="https://bioblend.readthedocs.io/en/latest/api_docs/galaxy/all.html#bioblend.galaxy.jobs.JobsClient.get_metrics">Biobland </a></p>

<h3 id="galaxy-results"><span class="notranslate">Galaxy</span> results</h3>

<p>We launched a Jupyter notebook to analyse results we got and reproduce graphs from CAMI2 paper for 4 tools (Flye, Megahit, Abyss, MetaSPAdes) and additional GSA in order to compare our results with Gold Standard Assembly. Further steps for benchmarking analysis could be to plot both individual assembly quast results and co-assembly quast results.</p>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial we analysed all CAMI2 challenges and reproduced the assembly challenge on a marine dataset to demonstrate that <span class="notranslate">Galaxy</span> is a viable solution for such challenges.</p>

<p>After reviewing all CAMI2 datasets, we selected the marine dataset, which is the most suitable for our purposes.</p>

<p>In <span class="notranslate">Galaxy</span>, we used the dataset collection function and the FASTQ splitter tool to preprocess the data.</p>

<p>The FASTQ splitter tool has been updated to improve its output names.</p>

<p>Among the assembly tools used in CAMI2, we selected Flye, Megahit, MetaSPAdes, and Abyss as being the most relevant for our needs. With these tools, you can perform short, long, or hybrid <span class="notranslate">reads</span>, which is ideal for this demonstration.</p>

<p>The versions of tools used in CAMI2 were compared with those in <span class="notranslate">Galaxy</span>.</p>

<p>We updated tools of outdated versions and with lack of some functions we needed to reproduce CAMI2.</p>

<p>Different sets of parameters were used for each tool to analyse outputs and draw conclusions about the most appropriate parameters.</p>

<p>We analysed benchmarking statistics from CAMI2 and chose the most appropriate for our goal.</p>

<p>Quast tool was used to analyse results from assembly tools. Additionally, we used bowtie2 tool and Map with minimap2 and Samtools Stat tool to obtain information about the percentage of <span class="notranslate">reads</span> mapped in assembly.</p>

<p>MultiQC and Jupiter notebook were used to create final reports and visualise results.</p>

<p>The datasets were uploaded to <span class="notranslate">Galaxy</span> using an FTP server.</p>

<p>During the project, we created <span class="notranslate">galaxy</span> histories that are available to the public. They are slightly different from those we suggest in this tutorial. We created them according to the principle of one history per:</p>
<ul>
  <li>Long <span class="notranslate">reads</span> or short <span class="notranslate">reads</span> as input</li>
  <li>Co-assembly or individual assembly</li>
  <li>one unique tool version</li>
  <li>one unique set of parameters</li>
</ul>

<blockquote class="details">
  <h3 id="details-details-of-galaxy-histories"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Details of <span class="notranslate">Galaxy</span> histories</h3>

  <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=915388998&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>

</blockquote>

<p>Furthermore, we have created this tutorial for those who are interested in such a topic and would like to reproduce CAMI2 assembly challenge on marine datasets. This tutorial could be undoubtedly useful for further improvement.</p>

<p>The further improvement of this tutorial could be:</p>

<ol>
  <li><span class="notranslate">Galaxy</span> workflows could be shared via IWC or Dockstore for the reproducibility</li>
  <li>Data could be formatted and submitted directly CAMI benchmarking portal via <span class="notranslate">Galaxy</span></li>
  <li>Other CAMI datasets could be used similarly</li>
  <li>Other CAMI challenges could be run similarly</li>
</ol>

<p>Ultimately, the project demonstrated that <span class="notranslate">Galaxy</span> could be used for CAMI and similar challenges. Aside from its impressive features, user-friendly interface, and ample resources, the platform has significant potential for future advancement.</p>

<p>Well done! <i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span></p>


                
                <blockquote class="key_points">
                    <div class="box-title" aria-label="key-points box"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Galaxy could be a good solution to reproduce CAMI</p>
</li>
                        
                        <li><p>CAMI2 consists of 4 challenges: assembly, genome binning, taxonomix binning, taxonomic profiling</p>
</li>
                        
                        <li><p>CAMI2 provides 5 datasets: pathogen detection, marine, strain madness, plant-associated</p>
</li>
                        
                        <li><p>Short reads and long reads can be assembled with different tools</p>
</li>
                        
                        <li><p>Co-assembly and individual assembly should be performed for dataset</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="noauthor_critical_nodate"><b>Critical Assessment of Metagenome Interpretation—a benchmark of metagenomics software \textbar Nature Methods</b>. <a href="https://www.nature.com/articles/nmeth.4458">https://www.nature.com/articles/nmeth.4458</a></li>
<li id="noauthor_critical_nodate-1"><b>Critical Assessment of Metagenome Interpretation - the second round of challenges \textbar bioRxiv</b>. <a href="https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1.full">https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1.full</a></li>
<li id="noauthor_galaxy_nodate-1"><b>Galaxy Training: Using dataset collections</b>. <a href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html</a></li>
<li id="noauthor_building_nodate"><b>Building Galaxy Tools Using Planemo — Planemo 0.75.0.dev0 documentation</b>. <a href="https://planemo.readthedocs.io/en/latest/writing_standalone.html">https://planemo.readthedocs.io/en/latest/writing_standalone.html</a></li>
<li id="noauthor_galaxy_nodate-4"><b>Galaxy Training: Assembly</b>. <a href="https://training.galaxyproject.org/training-material/topics/assembly/">https://training.galaxyproject.org/training-material/topics/assembly/</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy (Metagenomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            PlushZ, Bérénice Batut,  <b>Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/cami2-assembly/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/cami2-assembly/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false" aria-label="Toggle details box: BibTeX for citing this tutorial">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{metagenomics-cami2-assembly,
author = "PlushZ and Bérénice Batut",
title = "Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy (Galaxy Training Materials)",
year = "",
month = "",
day = ""
url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/cami2-assembly/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
        <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

        <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
        <script type="text/javascript">
        var snippets=document.querySelectorAll('div.highlight');
        [].forEach.call(snippets,function(snippet){
            snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });

        var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
            target:function(trigger){return trigger.nextElementSibling;
        }});
        </script>
        

        <script type="text/javascript">
            if(window.location.hostname === "galaxyproject.github.io") {
                // Redirect
                var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
                $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

                window.setTimeout(function(){
                    window.location.href = redirect;
                }, 5000)

            }
        </script>
    </body>
</html>
